src/server/services/wfs/qgswfsutils.h

```diff
@@ -58,9 +58,9 @@ namespace QgsWfs
   QgsVectorLayer *layerByTypeName( const QgsProject *project, const QString &typeName );
 
   /**
-   * Transform a Filter element to a feature request
+   * Transform a Filter element to a feature request and update server feature ids
    */
-  QgsFeatureRequest parseFilterElement( const QString &typeName, QDomElement &filterElem, const QgsProject *project = nullptr );
+  QgsFeatureRequest parseFilterElement( const QString &typeName, QDomElement &filterElem, const QgsProject *project = nullptr, QStringList *serverFids = new QStringList() );
```

Comment:
```suggestion
  QgsFeatureRequest parseFilterElement( const QString &typeName, QDomElement &filterElem, const QgsProject *project = nullptr, const QStringList &serverFids = QStringList() );
```

Reply:
serverFids cannot be a const. parseFilterElement adds serverFid found in filter.

Reply:
in this case use `nullptr` as default and properly check for that as proposed by @pblottiere

Reply:
I have updated the method to use const QStringList for serverFids.

Reviews are welcome.